spec/account/account-service.spec.js

```diff
@@ -25,31 +25,40 @@ describe("AccountService", () => {
             return promiseMock.resolve(true);
         });
         config = {
+            account: {
+                accountDir: "some/place/",
+                idHash: {
+                    algo: "sha256",
+                    hashLength: 24,
+                    iterations: 10000,
+                    salt: "pinkFullyUnicornsDancingOnRainbows"
+                }
+            },
             storage: {
                 bucket: "some-place-wonderful"
             }
         };
-        password = jasmine.createSpyObj("password", [
-            "hashContent"
+        secureHash = jasmine.createSpyObj("secureHash", [
+            "hashAsync"
         ]);
-        password.hashContent.andCallFake(() => {
-            return "hashedContent";
+        secureHash.hashAsync.andCallFake(() => {
+            return promiseMock.resolve("hashedContent");
         });
-        accountService = new AccountService(config, password, storageFake);
+
+        accountService = new AccountService(config, secureHash, storageFake);
     });
     describe(".completeAsync()", () => {
         it("puts the information successfully", (done) => {
             storageFake.getAsync.andCallFake(() => {
-                return promiseMock.resolve('{"email": "some.one@example.net"}');
+                return promiseMock.resolve('{"accountId": "unhashedAccountId", "email": "some.one@example.net"}');
             });
             accountService.completeAsync("directory", {
                 password: "somereallylonghashedpassword"
             }, {
                 expires: new Date()
             }).then((result) => {
                 expect(result).toEqual({
-                    email: "some.one@example.net",
-                    password: "somereallylonghashedpassword"
+                    accountId: "unhashedAccountId"
```

Comment:
This should be returning the hashed account ID, not the unhashed.  We supplied the unhashed one, yes?  Or pulled it from the storage mechanism?